src/EFCore/ChangeTracking/Internal/RelationshipsSnapshot.cs

```diff
@@ -41,13 +43,13 @@ public void SetValue(IPropertyBase propertyBase, object value)
 
                 Check.DebugAssert(!IsEmpty, "relationship snapshot is empty");
                 Check.DebugAssert(
-                    !(propertyBase is INavigation) || !((INavigation)propertyBase).IsCollection,
+                    propertyBase is not INavigation navigation || !navigation.IsCollection,
```

Comment:
nit: pattern matching FTW:

```suggestion
                    propertyBase is not INavigation { IsCollection: true },
```